packages/form-builder/src/blocks/fields/amount/Edit.tsx

```diff
@@ -1,51 +1,154 @@
+    const RecurringPeriod = ({count}) => {
+        const interval = count ?? recurringBillingInterval;
+
+        const singular = !isRecurringDonor
+            ? periodLookup[recurringBillingPeriod].singular
+            : periodLookup[recurringOptInDefaultBillingPeriod ?? 'month'].singular;
+
+        const plural = !isRecurringDonor
+            ? periodLookup[recurringBillingPeriod].plural
+            : periodLookup[recurringOptInDefaultBillingPeriod ?? 'month'].plural;
+
+        return (
+            <strong>
+                {1 === interval && <>{singular}</>}
+                {1 !== interval && (
+                    <>
+                        {interval} {plural}
+                    </>
+                )}
+            </strong>
+        );
+    };
+
+    const RecurringMessage = () => {
+
+        const interval = parseInt(recurringBillingInterval);
+
+        const periodPlaceholder = <strong>{'[' + _n('period', 'periods', interval, 'give') + ']'}</strong>;
+
+        const count = parseInt(recurringLengthOfTime);
+        const countElement = <strong>{count}</strong>;
+
+        const translatableString = !count
+            ? __('This donation occurs every <period />.', 'give')
+            : __('This donation occurs every <period /> for <count /> <strong>payments</strong>.', 'give');
+
+        return (
+            <Notice>
+                { 'one-time' === recurringOptInDefaultBillingPeriod
+                    ? __('You have chosen to make this donation one-time.', 'give')
+                    : createInterpolateElement(translatableString, {
+                        period: <RecurringPeriod count={interval} />,
+                        count: countElement,
+                        strong: <strong />, // Required to interpolate the strong tag around the "payments" text.
+                    })
+                }
+            </Notice>
+        );
+    };
+
+    const FixedRecurringMessage = () => (
+        <Notice>
+            {createInterpolateElement(__('This donation is <amount/> every <period/>. ', 'give'), {
+                amount: <strong>{amountFormatted}</strong>,
+                period: <RecurringPeriod count={parseInt(recurringBillingInterval)} />,
+            })}
+        </Notice>
+    );
+
+    const BillingPeriodControl = ({options}) => {
+        const [option, setOption] = useState(options[0]);
+        return (
+            <RadioControl
+                className={'give-billing-period-control'}
+                label={__('Billing Period', 'give')}
+                hideLabelFromVision={true}
+                selected={option}
+                options={['one-time'].concat(options).map((option) => {
+                    return {
+                        label: 'one-time' === option ? __('One Time', 'give') : periodLookup[option].adjective,
+                        value: option,
+                    };
+                })}
+                onChange={(value) => setOption(value)}
+            />
+        );
```

Comment:
1. Do we even need an onChange here? i'm not sure it's doing anything for the UI
2. Let's update the default selected option to be the default opt in period
3. Let's give the selected value a css border color so it's more clear what is selected as the default (with our recent recurring settings being more focused on that).

```
.give-billing-period-control .components-radio-control__input[type=radio] + label {
    border: 1px solid transparent;
    border-radius: 2px;
}

.give-billing-period-control .components-radio-control__input[type=radio]:checked + label {
    border-color: black;
}
```

<img width="1091" alt="Screenshot 2023-03-09 at 11 34 01 AM" src="https://user-images.githubusercontent.com/10138447/224090043-825ec438-ca67-493b-920b-f305f872fb23.png">

Going along with my comment about updating the conditional messaging display, this small UI update removes the need to announce what the selection is as it's visually intuitive.

Reply:
Resolved, see https://github.com/impress-org/givewp-next-gen/pull/128/commits/6019e3eb5d24f5d36ac935b9019d0cf1bb69551a.